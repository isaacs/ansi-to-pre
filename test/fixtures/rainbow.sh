#!/usr/bin/env bash
seq=$'\u2588\u2588\u2593\u2593\u2592\u2592\u2591\u2591'

r=(1 3 2 6 4 5)
rainbow () {
  f=$1
  b=$2
  t=$3
  echo -n $'\x1b['$t';30;100m'$seq
  for i in {0..6}; do
    fc=$([ $i -eq 0 ] && echo 90 || echo "$f${r[$[$i - 1]]}")
    bc=$([ $i -eq 6 ] && echo 100 || echo "$b${r[$i]}")
    echo -n $'\x1b['"${fc};${bc}m$seq"
  done
  echo -n $'\x1b[90;40m'$seq
  echo $'\x1b[40m  \x1b[0;22m'
}
rainbow 3 4 2
rainbow 3 4
rainbow 9 10
rainbow 9 10 2
