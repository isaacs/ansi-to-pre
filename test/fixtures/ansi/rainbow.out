]8;;\[0;38;2;0;0;0;48;2;64;64;64;2mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘]8;;\[0;38;2;64;64;64;48;2;255;0;0;2mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘]8;;\[0;38;2;255;0;0;48;2;224;224;0;2mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘]8;;\[0;38;2;224;224;0;48;2;0;160;0;2mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘]8;;\[0;38;2;0;160;0;48;2;0;176;176;2mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘]8;;\[0;38;2;0;176;176;48;2;0;64;224;2mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘]8;;\[0;38;2;0;64;224;48;2;224;0;224;2mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘]8;;\[0;38;2;224;0;224;48;2;64;64;64;2mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘]8;;\[0;38;2;64;64;64;48;2;0;0;0;2mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘  
]8;;\[0;38;2;0;0;0;48;2;64;64;64mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘]8;;\[0;38;2;64;64;64;48;2;255;0;0mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘]8;;\[0;38;2;255;0;0;48;2;224;224;0mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘]8;;\[0;38;2;224;224;0;48;2;0;160;0mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘]8;;\[0;38;2;0;160;0;48;2;0;176;176mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘]8;;\[0;38;2;0;176;176;48;2;0;64;224mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘]8;;\[0;38;2;0;64;224;48;2;224;0;224mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘]8;;\[0;38;2;224;0;224;48;2;64;64;64mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘]8;;\[0;38;2;64;64;64;48;2;0;0;0mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘  
]8;;\[0;38;2;0;0;0;48;2;64;64;64mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘]8;;\[0;38;2;64;64;64;48;2;255;48;48mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘]8;;\[0;38;2;255;48;48;48;2;255;255;0mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘]8;;\[0;38;2;255;255;0;48;2;0;255;0mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘]8;;\[0;38;2;0;255;0;48;2;0;255;255mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘]8;;\[0;38;2;0;255;255;48;2;0;128;255mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘]8;;\[0;38;2;0;128;255;48;2;255;51;255mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘]8;;\[0;38;2;255;51;255;48;2;64;64;64mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘]8;;\[0;38;2;64;64;64;48;2;0;0;0mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘  
]8;;\[0;38;2;0;0;0;48;2;64;64;64;2mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘]8;;\[0;38;2;64;64;64;48;2;255;48;48;2mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘]8;;\[0;38;2;255;48;48;48;2;255;255;0;2mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘]8;;\[0;38;2;255;255;0;48;2;0;255;0;2mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘]8;;\[0;38;2;0;255;0;48;2;0;255;255;2mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘]8;;\[0;38;2;0;255;255;48;2;0;128;255;2mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘]8;;\[0;38;2;0;128;255;48;2;255;51;255;2mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘]8;;\[0;38;2;255;51;255;48;2;64;64;64;2mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘]8;;\[0;38;2;64;64;64;48;2;0;0;0;2mâ–ˆâ–ˆâ–“â–“â–’â–’â–‘â–‘  [m